Dockerfile

```diff
@@ -1,47 +1,48 @@
-FROM nginx
-
-#Setting up Dockerfile to install the version of Converse to retrieve
-
-ENV CONVERSEJS_VERSION v8.0.1
-ENV CONVERSEJS_VERSION_S 8.0.1
-
-# Updating package lists then install Node.js and NPM for development purpose
-
-RUN apt-get update -y
-RUN apt-get install -y curl python-twisted python-openssl sudo unzip git apt-utils 
-RUN curl -sL https://deb.nodesource.com/setup_16.x | sudo -E bash -
-RUN sudo apt-get install -y nodejs
-RUN sudo apt-get install -y build-essential
-
-#Retrieving the version of Converse setup earlier
-
-RUN curl -L -O https://github.com/conversejs/converse.js/archive/${CONVERSEJS_VERSION}.zip \
-    && unzip ${CONVERSEJS_VERSION}.zip \
-    && rm -f ${CONVERSEJS_VERSION}.zip
-
-#Moving Converse content in the correct directory and making the teamchat page the main one
-
-RUN mv -v /converse.js-${CONVERSEJS_VERSION_S}/* /usr/share/nginx/html/
-RUN rm -rf /converse.js-${CONVERSEJS_VERSION_S}/
-RUN cd /usr/share/nginx/html/ && make dev 
-RUN mv /usr/share/nginx/html/index.html /usr/share/nginx/html/index_bak.html \
-    && cp /usr/share/nginx/html/fullscreen.html /usr/share/nginx/html/index.html 
-
+FROM nginx:latest
+
+LABEL maintainer="iam@f4b.io"
+
+ARG BUILD_VERSION
+ARG BUILD_DATE
+ARG DEBIAN_FRONTEND=noninteractive
+ARG CONVERSEJS_VERSION=9.1.0
+
+LABEL org.label-schema.schema-version="9.1.0"
+LABEL org.label-schema.name="converse.js/converse.js"
+LABEL org.label-schema.description="messaging freedom"
+LABEL org.label-schema.url="https://conversejs.org/"
+LABEL org.label-schema.application="converse.js"
+LABEL org.label-schema.build-date=$BUILD_DATE
+LABEL org.label-schema.version=$BUILD_VERSION
+LABEL org.opencontainers.image.source="https://github.com/f4bio/converse.js"
+
+WORKDIR /workspace
+
+# # Replace nginx default configuration file
+COPY default.conf /etc/nginx/conf.d/default.conf
+
+# Update deps, install required & additional libraries
+RUN apt-get --yes update
+RUN apt-get --yes install \
+  software-properties-common gnupg gosu curl ca-certificates zip unzip git
+RUN apt-get --yes install python3-dev python3-twisted python3-openssl
+
+RUN curl -sLO \
+  https://github.com/conversejs/converse.js/releases/download/v$CONVERSEJS_VERSION/converse.js-$CONVERSEJS_VERSION.tgz
+RUN tar xzf converse.js-$CONVERSEJS_VERSION.tgz
+RUN mv package/* /usr/share/nginx/html/
+RUN curl -sL \
+  -o /usr/share/nginx/html/index.html \
+  https://raw.githubusercontent.com/conversejs/converse.js/master/fullscreen.html
```

Comment:
The file `fullscreen.html` being downloaded is hardcoded to use the hosted Punjab instance at https://conversejs.org/http-bind/ (see the `bosh_service_url` setting).

Given that this docker build is including its own Punjab instance, we'll need to use an index.html file that sets `bosh_service_url` to the URL of this build's Punjab instance.

Reply:
oh. nice! I didn't notice that at all. 
I'm not really a fan of search&replace string in files like this, so I hope this solution is fine by you too?!